hazelcast/src/main/java/com/hazelcast/internal/monitors/HealthMonitor.java

```diff
@@ -69,227 +58,415 @@
-                }
+        int delaySeconds = node.getGroupProperties().HEALTH_MONITORING_DELAY_SECONDS.getInteger();
+        return new HealthMonitorThread(delaySeconds);
+    }
+
+    public HealthMonitor start() {
+        if (monitorLevel == OFF) {
+            logger.finest("HealthMonitor is disabled");
+            return this;
+        }
+
+        monitorThread.start();
+        logger.finest("HealthMonitor started");
+        return this;
+    }
+
+    private HealthMonitorLevel getHealthMonitorLevel() {
+        GroupProperties properties = node.getGroupProperties();
+        String healthMonitorLevelString = properties.HEALTH_MONITORING_LEVEL.getString();
+        return valueOf(healthMonitorLevelString);
+    }
 
-                try {
-                    Thread.sleep(TimeUnit.SECONDS.toMillis(delaySeconds));
-                } catch (InterruptedException e) {
-                    return;
+    private final class HealthMonitorThread extends Thread {
+        private final int delaySeconds;
+        private boolean performanceLogHint;
+
+        private HealthMonitorThread(int delaySeconds) {
+            super(node.getHazelcastThreadGroup().getInternalThreadGroup(),
+                    node.getHazelcastThreadGroup().getThreadNamePrefix("HealthMonitor"));
+            setDaemon(true);
+            this.delaySeconds = delaySeconds;
+            this.performanceLogHint = node.getGroupProperties().PERFORMANCE_MONITOR_ENABLED.getBoolean();
+        }
+
+        @Override
+        public void run() {
+            HealthGaugesRenderer renderer = new HealthGaugesRenderer();
+
+            try {
+                while (node.isActive()) {
+                    renderer.init();
+                    switch (monitorLevel) {
+                        case NOISY:
+                            logger.log(Level.INFO, renderer.render());
+                            break;
+                        case SILENT:
+                            if (renderer.exceedsThreshold()) {
+                                logPerformanceMonitorHint();
+                                logger.log(Level.INFO, renderer.render());
+                            }
+                            break;
+                        default:
+                            throw new IllegalStateException("unrecognized HealthMonitorLevel:" + monitorLevel);
+                    }
+
+                    try {
+                        Thread.sleep(SECONDS.toMillis(delaySeconds));
+                    } catch (InterruptedException e) {
+                        return;
+                    }
                 }
+            } catch (OutOfMemoryError e) {
+                OutOfMemoryErrorDispatcher.onOutOfMemory(e);
+            } catch (Throwable t) {
+                logger.warning("Health Monitor failed", t);
             }
-        } catch (OutOfMemoryError e) {
-            OutOfMemoryErrorDispatcher.onOutOfMemory(e);
+        }
+
+        private void logPerformanceMonitorHint() {
```

Comment:
I guess the hint is appropriate here, but why it's called only when the monitor level is set to SILENT?